### ISSUE TITLE
Scope monorail service to track traditional and semantic search use

### ISSUE THREAD

**Issue by yurm04** [Shopify/polaris-react-archive#9084]
TL;DR we want to set up a private backend service to make calls to Monorail to track events on polaris.shopify.com

## **Context**
As part of the AI search MVP we wanted to be able to establish a baseline understanding of how our current Global Search feature is being used. This will allow us to compare our Global Search performance with the new AI semantic search feature. It will also help inform decisions on future shopify.dev projects related to AI search features.

### **Blocked by GA**
To start tracking our Global Search, I tried to set up custom Google Analytics events. However, we're blocked with GA because the GA support team needs to set up custom metric views for polaris.shopify.com and [they don't have the bandwidth to take on this work for at least a few weeks](https://shopify.slack.com/archives/C042QM635/p1682431256587129?thread_ts=1678315153.981959&cid=C042QM635). Additionally, our current version of GA will be deprecated on July 1 and the support team likely won't have the bandwidth to help us migrate to GA4.

### **Blocked by Monorail**
We also attempted to [set up Monorail to work with Global Search](https://github.com/Shopify/polaris/pull/8702). This approach was blocked when we learned that we [do not want to call Monorail directly](https://discourse.shopify.io/t/using-private-shopify-npm-packages-in-public-repo/37521).

## **Proposed solution**
Using Monorail as our data platform is the best solution in the long term because we will be able to directly set up and compare analytics with shopify.dev. In order to do this, we will need to set up a private backend service with Monorail installed that can be called directly from polaris.shopify.com with data events. Keeping the backend service private would address security concerns of including Monorail in a public repository

<img width="1367" alt="Screenshot 2023-04-25 at 1 53 39 PM" src="https://user-images.githubusercontent.com/4642404/234375061-f1a4c9c0-4176-4e0f-8462-c1ce3e425b98.png">

In addition to forwarding/tracking search events with this monorail service, we can also track other events that GA has blocked us from setting up. This will eliminate the need to continue using GA and migrate to GA4 and prevent any blackouts in our analytics starting in July when our version of GA is deprecated.

## **Open questions**
* What is the effort to spin up this backend service?
* Maintenance burden of creating a new service?
* Is there an existing private service we can leverage instead of creating a net new service?

**Comment by kyledurand** [comment]
I'm down for spinning up a private service if it unblocks us.

> What is the effort to spin up this backend service?
> Maintenance burden of creating a new service?
> Is there an existing private service we can leverage instead of creating a net new service?

I think the first two are fairly minimal. Dependabot should be able to merge security upgrades automatically and we'd just have to worry about breaking changes to monorail, something we'd have to do anyway. I can't think of any new service we could use here and monorail is what we use across the company anyway.

**Comment by alex-page** [comment]
We definitely want to be careful adding new services. They must be maintained and this would have to be tier 3 and that requires a security audit. Is there any way that `polaris-site-container-builder` could help here? That is our website service that clones the `polaris` monorepo.
